app/screens/interactive_dialog/dialog_element.test.js

```diff
@@ -36,4 +37,49 @@ describe('DialogElement', () => {
         );
         expect(wrapper.find({secureTextEntry: false}).exists()).toBe(true);
     });
-});
\ No newline at end of file
+
+    describe('radioSetting', () => {
+        const radioOptions = [
+            {value: 'foo', text: 'foo-text'},
+            {value: 'bar', text: 'bar-text'},
+        ];
+
+        test('RadioSetting is rendered when type is radio', () => {
+            const wrapper = shallow(
+                <DialogElement
+                    {...baseDialogProps}
+                    theme={theme}
+                    type='radio'
+                    options={radioOptions}
+                />
+            );
+
+            expect(wrapper.find(RadioSetting).exists()).toBe(true);
+        });
+
+        test('The default value is the first element of the list', () => {
+            const wrapper = shallow(
+                <DialogElement
+                    {...baseDialogProps}
+                    theme={theme}
+                    type='radio'
+                    options={radioOptions}
+                />
+            );
+            expect(wrapper.find({values: radioOptions, value: radioOptions[0].value}).exists()).toBe(true);
+        });
+
+        test('The default value can be specified from the list', () => {
+            const wrapper = shallow(
+                <DialogElement
+                    {...baseDialogProps}
+                    theme={theme}
+                    type='radio'
+                    options={radioOptions}
+                    value={radioOptions[1].value}
+                />
+            );
+            expect(wrapper.find({values: radioOptions, value: radioOptions[1].value}).exists()).toBe(true);
+        });
```

Comment:
what would happen if we specify a value not on the list? since it's not an index of the list but the values themselves

Reply:
Then the component still renders, but without any option selected. The value is still assigned to the option that is not in the list.

In this case, we can 1. default it to the first element of the given options or 2. display it as an error and let end users to select valid option from the list. Any suggestions on this?

Reply:
I'd prefer 2, as this will cause less confusion (e.g.: a typo and we get a different value)

Reply:
https://github.com/mattermost/mattermost-redux/pull/916

Reply:
Said PR has been merged.  Please update package.json, thanks